tests/components/essent/test_sensor.py

```diff
@@ -0,0 +1,445 @@
+"""Test the Essent sensors."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+from unittest.mock import AsyncMock
+
+from freezegun.api import FrozenDateTimeFactory
+import pytest
+
+from essent_dynamic_pricing.models import EnergyData, EssentPrices
+from homeassistant.const import STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util import dt as dt_util
+
+from homeassistant.components.essent.sensor import (
+    _format_dt_str,
+    _parse_tariff_datetime,
+)
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+from . import setup_integration
+
+pytestmark = [
+    pytest.mark.usefixtures("entity_registry_enabled_by_default"),
+    pytest.mark.freeze_time("2025-11-16 10:30:00+01:00"),
+]
+
+
+async def _enable_unique_ids(
+    hass: HomeAssistant,
+    entry: MockConfigEntry,
+    ent_reg: er.EntityRegistry,
+    unique_ids: tuple[str, ...],
+) -> None:
+    """Enable disabled entities and reload when needed."""
+    updated = False
+    for unique_id in unique_ids:
+        entity_id = ent_reg.async_get_entity_id("sensor", "essent", unique_id)
+        assert entity_id is not None
+        reg_entry = ent_reg.async_get(entity_id)
+        assert reg_entry is not None
+        if reg_entry.disabled_by:
+            ent_reg.async_update_entity(entity_id, disabled_by=None)
+            updated = True
+
+    if updated:
+        await hass.config_entries.async_reload(entry.entry_id)
+        await hass.async_block_till_done()
+
+
+async def test_sensor_states(hass: HomeAssistant) -> None:
+    """Test the sensor states and attributes."""
+    entry = await setup_integration(hass)
+    ent_reg = er.async_get(hass)
+    await _enable_unique_ids(
+        hass,
+        entry,
+        ent_reg,
+        ("electricity_next_price", "gas_next_price"),
+    )
+
+    def _state(unique_id: str):
+        entity_id = ent_reg.async_get_entity_id("sensor", "essent", unique_id)
+        assert entity_id is not None
+        state = hass.states.get(entity_id)
+        assert state is not None
+        return state
+
+    elec_current = _state("electricity_current_price")
+    elec_next = _state("electricity_next_price")
+    gas_current = _state("gas_current_price")
+    gas_next = _state("gas_next_price")
+
+    assert elec_current is not None
+    assert float(elec_current.state) == 0.25
+    assert elec_current.attributes["unit_of_measurement"] == "€/kWh"
+
+    assert elec_next is not None
+    assert float(elec_next.state) == 0.22
+    assert elec_next.attributes["unit_of_measurement"] == "€/kWh"
+
+    assert gas_current is not None
+    assert float(gas_current.state) == 0.82
+    assert gas_current.attributes["unit_of_measurement"] == "€/m³"
+
+    assert gas_next is not None
+    assert float(gas_next.state) == 0.78
+    assert gas_next.attributes["unit_of_measurement"] == "€/m³"
```

Comment:
Instead, look into `snapshot_platform`. It will snapshot all the entities when run with the snapshot-platform flag